### ISSUE TITLE
Take forever to install package

### ISSUE THREAD

**Issue by sucrecacao** [NixOS/nix#7469]
## Platform

- [ X] Linux: Qubes OS, debian VM
- [ ] macOS
- [ ] WSL

## Additional information

When install a package with `nix-env -iA ...` it takes forever to install the package ( didn't finished after hours) whereas it only take a few seconds with apt. 

I had this issue first when installing gcc, where I stopped and restarted the install process a few time as it was taking too much time. Now it seems that it does it for every install. 

## Output

I'm not going to print the whole enormous log but a lot the lines start with `gcc` or `checking...` even if I install a python based package

## Priorities

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by bjornfor** [comment]
What's your nixpkgs? The `staging` branch? Or do you have an overlay that changes stdenv?

On my system jq installed in less than 2 seconds:

```
$ nix-env -iA jq -f '<nixpkgs>'
installing 'jq-1.6'
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/pg73qqxg9ysnjbl3c0l0xvgv8c6n7q69-jq-1.6
copying path '/nix/store/pg73qqxg9ysnjbl3c0l0xvgv8c6n7q69-jq-1.6' from 'https://cache.nixos.org'...
building '/nix/store/bx8lv81c7nw96a5kvnzn3mxj22ambj89-user-environment.drv'...
```

It sounds like Nix started building things from source (`a lot the lines start with gcc or checking...`), and unless you asked for it, that sounds like an error.

**Comment by bjornfor** [comment]
kakoune installs in less than 2 seconds as well:

```
$ nix-env -iA kakoune -f '<nixpkgs>'
installing 'kakoune-2021.11.08'
this derivation will be built:
  /nix/store/sj6bghz57yq5kcd1q7ry9qd6hxcf0cpp-kakoune-2021.11.08.drv
this path will be fetched (1.03 MiB download, 4.56 MiB unpacked):
  /nix/store/59pmfbg9n0grmib5g3sak5xfja2kfxqz-kakoune-unwrapped-2021.11.08
copying path '/nix/store/59pmfbg9n0grmib5g3sak5xfja2kfxqz-kakoune-unwrapped-2021.11.08' from 'https://cache.nixos.org'...
building '/nix/store/sj6bghz57yq5kcd1q7ry9qd6hxcf0cpp-kakoune-2021.11.08.drv'...
building '/nix/store/5kynsbmhhy8c9jycqgqaa33hkkvlbybd-user-environment.drv'...
```

**Comment by rapenne-s** [comment]
From your information, it looks like what you are installing isn't in the binary cache, so nix has to compile it locally.

**Comment by sucrecacao** [comment]
Thanks @rapenne-s  and @bjornfor  for your quick answer 


I have solved the issue by deleting the old TemplateVM where nix was installed ( I tried to reinstall nix on the VM but it was just creating more trouble ). 
In a fresh new debian TemplateVM I have install nix **with the --no-daemon** option instead. And now it's working as expected, package are installed super quickly. 


I have no idea was caused the issue in the first place, but it might be QubesOS related.
